optaplanner-benchmark/src/main/java/org/optaplanner/benchmark/impl/statistic/movecountperstep/MoveCountPerStepSubSingleStatistic.java

```diff
@@ -21,52 +21,38 @@
 import org.optaplanner.benchmark.config.statistic.ProblemStatisticType;
 import org.optaplanner.benchmark.impl.result.SubSingleBenchmarkResult;
 import org.optaplanner.benchmark.impl.statistic.ProblemBasedSubSingleStatistic;
+import org.optaplanner.benchmark.impl.statistic.StatisticRegistry;
 import org.optaplanner.core.api.solver.Solver;
-import org.optaplanner.core.impl.localsearch.scope.LocalSearchStepScope;
-import org.optaplanner.core.impl.phase.event.PhaseLifecycleListenerAdapter;
-import org.optaplanner.core.impl.phase.scope.AbstractStepScope;
+import org.optaplanner.core.config.solver.metric.SolverMetric;
 import org.optaplanner.core.impl.score.definition.ScoreDefinition;
-import org.optaplanner.core.impl.solver.AbstractSolver;
+
+import io.micrometer.core.instrument.Tags;
 
 public class MoveCountPerStepSubSingleStatistic<Solution_>
         extends ProblemBasedSubSingleStatistic<Solution_, MoveCountPerStepStatisticPoint> {
 
-    private MoveCountPerStepSubSingleStatisticListener listener;
-
     public MoveCountPerStepSubSingleStatistic(SubSingleBenchmarkResult subSingleBenchmarkResult) {
         super(subSingleBenchmarkResult, ProblemStatisticType.MOVE_COUNT_PER_STEP);
-        listener = new MoveCountPerStepSubSingleStatisticListener();
     }
 
     // ************************************************************************
     // Lifecycle methods
     // ************************************************************************
 
     @Override
-    public void open(Solver<Solution_> solver) {
-        ((AbstractSolver<Solution_>) solver).addPhaseLifecycleListener(listener);
+    public void open(StatisticRegistry registry, Tags runTag, Solver<Solution_> solver) {
+        registry.addListener(SolverMetric.MOVE_COUNT_PER_STEP, timeMillisSpent -> {
+            registry.getGaugeValue(SolverMetric.MOVE_COUNT_PER_STEP + ".accepted", runTag, accepted -> {
+                registry.getGaugeValue(SolverMetric.MOVE_COUNT_PER_STEP + ".accepted", runTag, selected -> {
```

Comment:
I don't understand. Why the nesting? The gauge is the same, shouldn't therefore `accepted` and `selected` do the same thing?

Reply:
The guage should be different (it should be the selected gauge); good catch